ee_tests/src/specs/quickstart_cheterminal.spec.ts

```diff
@@ -0,0 +1,105 @@
+    support.debug('>>> Go to Reset Env Page - OK');
+
+    await cleanupEnvPage.cleanup(browser.params.login.user);
+  });
+
+  afterEach( async () => {
+    await browser.sleep(support.DEFAULT_WAIT);
+    support.writeScreenshot('target/screenshots/che_final_' + globalSpaceName + '.png');
+    support.info('\n ============ End of test reached ============ \n');
+    await dashboardPage.logout();
+  });
+
+  /* Accept all defaults for new quickstarts */
+
+    // tslint:disable:max-line-length
+  it('Create a new space, new ' + browser.params.quickstart.name + ' quickstart, create Che workspace, run maven in the workspace', async () => {
+    let quickstart = new Quickstart(browser.params.quickstart.name);
+    let spaceName = support.newSpaceName();
+    globalSpaceName = spaceName;
+    let spaceDashboardPage = await dashboardPage.createNewSpace(spaceName);
+
+    let wizard = await spaceDashboardPage.addToSpace();
+
+    support.info('Creating quickstart: ' + quickstart.name);
+    await wizard.newQuickstartProject({ project: quickstart.name });
+    await spaceDashboardPage.ready();
+
+    /* Open Che display page */
+    await spaceDashboardPage.codebasesSectionTitle.clickWhenReady();
+
+    let spaceChePage = new SpaceChePage();
+    await spaceChePage.createCodebase.clickWhenReady(support.LONGEST_WAIT);
+
+    /* A new browser window is opened when Che opens - switch to that new window now */
+    let handles = await browser.getAllWindowHandles();
+    support.debug('Number of browser tabs before = ' + handles.length);
+    await browser.wait(windowCount(2), support.DEFAULT_WAIT);
+    handles = await browser.getAllWindowHandles();
+
+    support.debug('Number of browser tabs after = ' + handles.length);
+    support.writeScreenshot('target/screenshots/che_workspace_parta_' + spaceName + '.png');
+
+    /* Switch to the Che browser window */
+    await browser.switchTo().window(handles[1]);
+
+    let spaceCheWorkSpacePage = new SpaceCheWorkspacePage();
+    support.writeScreenshot('target/screenshots/che_workspace_partb_' + spaceName + '.png');
+
+    let projectInCheTree = new Button(spaceCheWorkSpacePage.recentProjectRootByName(spaceName), 'Project in Che Tree');
+    await projectInCheTree.untilPresent(support.LONGEST_WAIT);
+    support.writeScreenshot('target/screenshots/che_workspace_partc_' + spaceName + '.png');
+
+    expect(await spaceCheWorkSpacePage.recentProjectRootByName(spaceName).getText()).toContain(spaceName);
+
+    /* Open the terminal window and execute command */
+    await spaceCheWorkSpacePage.bottomPanelTerminalTab.clickWhenReady();
+    await support.printTerminal(spaceCheWorkSpacePage, 'cd ' + spaceName);
+
+    await support.printTerminal(spaceCheWorkSpacePage, 'mvn clean install -Popenshift,openshift-it');
+    await browser.wait(until.textToBePresentInElement(spaceCheWorkSpacePage.bottomPanelTerminal, 'BUILD SUCCESS'), support.LONGER_WAIT);
```

Comment:
There should be some logic that will fail the test as soon as 'BUILD FAILURE' appears in console (as happened to me).

Reply:
Excellent idea!